core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/ProducerBindingService.java

```diff
@@ -79,14 +80,25 @@ public void delete(ProducerBinding producerBinding) {
   }
 
   public Iterable<ProducerBinding> findAll(Predicate<ProducerBinding> filter) {
-    return stream(producerBindingRepository.findAll().spliterator(), false)
-        .filter(r -> filter.test(r))
-        .collect(Collectors.toList());
+    return producerBindingRepository.findAll().stream().filter(filter).collect(toList());
```

Comment:
Might be possible to use the 'Example' process if the filter has any non-regex fields, then apply the Predicate for the regex. Something for tomorrow?

Reply:
Everything is a regex string apart from the stream version which is an integer. May be worth exploring but it's not low hanging fruit and also separately from this PR.

Reply:
Just putting it out there before I forget.